src/physics/euler.cpp

```diff
@@ -295,6 +295,22 @@ ::compute_specific_enthalpy ( const std::array<real,nstate> &conservative_soln,
     return specific_enthalpy;
 }
 
+template <int dim, int nstate, typename real>
+inline real Euler<dim,nstate,real>
+::compute_numerical_entropy_function ( const std::array<real,nstate> &conservative_soln ) const
+{
+    const real pressure = compute_pressure<real>(conservative_soln);
+    const real density = conservative_soln[0];
+
+    real entropy = pressure * pow(density, -gam);
+    if (entropy > 0)    entropy = log( entropy );
+    else                entropy = BIG_NUMBER;
```

Comment:
maybe put this in a function like I've done in PR #194

Reply:
that way you get a message if its negative

Reply:
That's a good suggestion. I've made that change.